src/main/java/com/maxdemarzi/shortest/Service.java

```diff
@@ -438,6 +405,59 @@ private void streamShortestPathsUsingHandwrittenBFS(String centerEmail, List<Str
         }
     }
 
+    private static final Map<String, Integer> relationshipCosts = ImmutableMap.<String, Integer>builder()
+        .put("EqualTo", 1)
+        .put("AuthoredBy", 1)
+        .put("CoAuthorOf", 1)
+        .put("ContainsEmail", 1)
+        .put("Follows", 1)
+        .put("hasContact", 1)
+        .put("HasEmail", 1)
+        .put("HasUrl", 1)
+        .build();
+
+    private void streamShortestPathsUsingDijkstra(String centerEmail, List<String> bibEntries, List<String> edgeEmails, int maxLength, JsonGenerator jg) {
+        final Long centerNodeId;
+        try {
+            centerNodeId = nodeCache.getEmailNode(centerEmail);
+        } catch (ExecutionException e) {
+            return;
+        }
+        Collection<Long> startNodes = nodeCache.getBibliographEntryNodes(bibEntries);
+        startNodes.add(centerNodeId);
+
+        final HashLongObjMap<String> edgeEmailsByNodeId = HashLongObjMaps.newMutableMap();
+        for (String edgeEmail : edgeEmails) {
+            try {
+                edgeEmailsByNodeId.put(nodeCache.getEmailNode(edgeEmail), edgeEmail);
+            } catch (Exception e) {
+                continue;
+            }
+        }
+
+        try (Transaction tx = db.beginTx()) {
+            ThreadToStatementContextBridge ctx = dbAPI.getDependencyResolver().resolveDependency(ThreadToStatementContextBridge.class);
+            ReadOperations ops = ctx.get().readOperations();
+
+            new Dijkstra(ops, relationshipCosts, startNodes, 4, new Dijkstra.NodeCallback() {
```

Comment:
i believe the existing endpoint takes a max path length as an argument, because for a weakly-matching set of nodes we only go up to a distance of 3